internal/watcher/instance_watcher_service.go

```diff
@@ -187,3 +231,61 @@ func compareInstances(oldInstances, instances []*v1.Instance) (newInstances, del
 
 	return newInstances, deletedInstances
 }
+
+func (iw *InstanceWatcherService) updateNginxInstanceRuntime(
```

Comment:
Should unit tests be added for these ? I think there was one in the resource plugin